### DISCUSSION TITLE
log uci output during call do `analyse`

### DISCUSSION THREAD

**Discussion by robertnurnberg** [niklasf/python-chess#1081] [Q&A] [answered]
Is it possible to obtain the UCI output from the thinking engine during a call to `analyse()`, before the method returns `info` once a limit is hit?

Edit: If this is not already implemented, then an option like `logFile=None` or `uciLogFile=None` to the engine constructor or the `analyse` function might be a possible way to provide these logs to the user. A common example would be something like a `go mate` command, where the user would like to see how far the engine is progressing before it manages to find the required mate score.

**Comment by niklasf** [comment, marked as answer]
Yes: https://python-chess.readthedocs.io/en/latest/engine.html#logging
